### PR TITLE
prerequsite: remove target_point_strings attribute

### PR DESCRIPTION
**Pull request by oliver-sanders**
The `Prerequisite` class contains two attributes with overlapping information, `target_point_strings` and `satisfied`.

See https://github.com/cylc/cylc-flow/pull/5466#discussion_r1178887978

Remove `target_point_strings` (only used internally) to save memory.

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: tests/unit/test_prerequisite.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        ('2000', 'b', 'succeeded'): 'satisfied naturally',
+        ('2000', 'c', 'succeeded'): 'satisfied naturally',
+        # the remaining dependency should be marked as forse-satisfied
+        ('2001', 'd', 'custom'): 'force satisfied',
+    }
+    # mark all prereqs as unsatisfied
+    prereq.set_not_satisfied()
+    assert prereq.satisfied == {
+        # all prereqs INCLUDING the pre-initial should be marked as nnot
+        # satisfied
```
Comment (oliver-sanders) [inline tests/unit/test_prerequisite.py:95]: Should the pre-initial dependency be set to `False` here?

It is initialised as `satisfied naturally` so it seems odd for this method to undo that.

File: cylc/flow/prerequisite.py [hunk 2]
```diff
@@ -41,9 +41,14 @@ class Prerequisite:
     """
 
     # Memory optimization - constrain possible attributes to this list.
-    __slots__ = ["satisfied", "_all_satisfied",
-                 "target_point_strings", "start_point",
-                 "conditional_expression", "point"]
+    __slots__ = (
+        "satisfied",
+        "_all_satisfied",
+        "target_point_strings",
```
Comment (MetRonnie) [inline cylc/flow/prerequisite.py:47]:
```suggestion
```

Comment (oliver-sanders) [reply cylc/flow/prerequisite.py:47]: Derp.

**Review by MetRonnie — APPROVED** [review]
Will conflict with #5504

File: tests/unit/test_prerequisite.py (hunk 1 above)
Comment (hjoliver) [reply tests/unit/test_prerequisite.py:92]: Maybe not, but I checked and it's actually not used anywhere in Cylc 8.  We should either remove it, or fix it (pre-initial) depending on the resolution of this open question: https://cylc.github.io/cylc-admin/proposal-cylc-set.html#un-setting-prerequisites

**Review by hjoliver — APPROVED** [review]

File: tests/unit/test_prerequisite.py (hunk 1 above)
Comment (hjoliver) [reply tests/unit/test_prerequisite.py:92]: I'll put up a new issue for this, as a reminder.

**Pull request merged**